packages/taquito/test/operations/proposals-operation.spec.ts

```diff
@@ -0,0 +1,74 @@
+import { ProposalsOperation } from '../../src/operations/proposals-operation';
+import { OperationContentsAndResult } from '@taquito/rpc';
+import { ForgedBytes } from '../../src/operations/types';
+import { defaultConfigConfirmation } from '../../src/context';
+
+describe('Proposals operation', () => {
+  let fakeContext: any;
+
+  const fakeForgedBytes = {} as ForgedBytes;
+
+  const successfulResult = [
+    {
+      kind: 'proposals',
+      source: 'tz1YPSCGWXwBdTncK2aCctSZAXWvGsGwVJqU',
+      period: 2,
+      proposals: ['PtKathmankSpLLDALzWw7CGD2j2MtyveTwboEYokqUCP4a1LxMg'],
+    },
+  ] as OperationContentsAndResult[];
+
+  beforeEach(() => {
+    fakeContext = {
+      rpc: {
+        getBlock: jest.fn(),
+      },
+      config: { ...defaultConfigConfirmation },
+    };
+
+    fakeContext.rpc.getBlock.mockResolvedValue({
+      operations: [[{ hash: 'ooBghN2ok5EpgEuMqYWqvfwNLBiK9eNFoPai91iwqk2nRCyUKgE' }], [], [], []],
+      header: {
+        level: 1,
+      },
+    });
+  });
+
+  it('should return source of Proposals operation', () => {
+    const op = new ProposalsOperation(
+      'ooBghN2ok5EpgEuMqYWqvfwNLBiK9eNFoPai91iwqk2nRCyUKgE',
+      {} as any,
+      'tz1YPSCGWXwBdTncK2aCctSZAXWvGsGwVJqU',
+      fakeForgedBytes,
+      successfulResult,
+      fakeContext
+    );
+
+    expect(op.source).toEqual('tz1YPSCGWXwBdTncK2aCctSZAXWvGsGwVJqU');
+  });
+
+  it('should return period of Proposals operation', () => {
+    const op = new ProposalsOperation(
+      'ooBghN2ok5EpgEuMqYWqvfwNLBiK9eNFoPai91iwqk2nRCyUKgE',
+      {} as any,
+      'tz1YPSCGWXwBdTncK2aCctSZAXWvGsGwVJqU',
+      fakeForgedBytes,
+      successfulResult,
+      fakeContext
+    );
+
+    expect(op.period).toEqual(2);
+  });
+
+  it('should return proposal hash of Proposals operation', () => {
+    const op = new ProposalsOperation(
+      'ooBghN2ok5EpgEuMqYWqvfwNLBiK9eNFoPai91iwqk2nRCyUKgE',
+      { proposals: ['PtKathmankSpLLDALzWw7CGD2j2MtyveTwboEYokqUCP4a1LxMg'] } as any,
```

Comment:
is this possible to be typed as something other than any?

Reply:
that's a good question, but I suspect the reason we use `any` for all of the params in the test is because we might or might not pass anything at all. So `any` gives us the flexibility. 

In general I'd stay away from the use of `any` as you're probably suggesting, but I think this usage in tests is probably okay.